### PR TITLE
fix end time schedule 

### PR DESCRIPTION
**Pull request by chengren311**
An issue was introduced in https://github.com/azkaban/azkaban/pull/1110 that causing schedule end time is set to a past date whenever SLA changes are made.
It's caused by incompatibility of new code https://github.com/azkaban/azkaban/pull/1110 with old schedule metadata. SLA change triggers reloading updated schedules into memory. But instead of using a default schedule end time, which we set to some day in year 2050, it always mistakenly uses a past date as the expiring time of the schedule. 
This fixes the issue by setting appropriate end time of the schedule.

### CODE REVIEW

**Comment by HappyRay** [comment]
Is it a clean revert?

If so, please indicate so in the commit message.

**Comment by reallocf** [comment]
This looks pretty good to me, but would like additional eyes @HappyRay

**Comment by chengren311** [comment]
@HappyRay no, this is a patch of fix.

File: azkaban-common/src/main/java/azkaban/scheduler/TriggerBasedScheduleLoader.java [hunk 1]
```diff
@@ -44,7 +44,7 @@
   private long lastUpdateTime = -1;
 
   public TriggerBasedScheduleLoader(final TriggerManager triggerManager,
-                                    final String triggerSource) {
+      final String triggerSource) {
```
Comment (HappyRay) [inline azkaban-common/src/main/java/azkaban/scheduler/TriggerBasedScheduleLoader.java]: This looks like a formatting change.  Is it intentional? I though the earlier code was formatted properly by the save action.

File: azkaban-common/src/main/java/azkaban/scheduler/TriggerBasedScheduleLoader.java [hunk 2]
```diff
@@ -174,6 +176,14 @@ private BasicTimeChecker getBasicTimeChecker(final Map<String, ConditionChecker>
     return null;
   }
 
+  private BasicTimeChecker getEndTimeChecker(final Trigger t) {
```
Comment (shuzhang1989) [inline azkaban-common/src/main/java/azkaban/scheduler/TriggerBasedScheduleLoader.java]: will t be null?

File: azkaban-common/src/main/java/azkaban/scheduler/TriggerBasedScheduleLoader.java (hunk 1 above)
Comment (chengren311) [inline azkaban-common/src/main/java/azkaban/scheduler/TriggerBasedScheduleLoader.java]: yes. previously code was not formatted properly.

File: azkaban-common/src/main/java/azkaban/scheduler/TriggerBasedScheduleLoader.java (hunk 2 above)
Comment (chengren311) [inline azkaban-common/src/main/java/azkaban/scheduler/TriggerBasedScheduleLoader.java]: nope

**Pull request merged**